src/QsCompiler/CompilationManager/FileContentManager.cs

```diff
@@ -305,20 +330,26 @@ private void InvalidateOrUpdateContextDiagnostics(int start, int count, int line
         /// Throws an ArgumentException if the given start and end position do not denote a valid range.
         /// </summary>
         private void InvalidateOrUpdateHeaderDiagnostics(Range syntaxCheckDelimiters, int lineNrChange) =>
-            this.InvalidateOrUpdateBySyntaxCheckDelimeters(this.HeaderDiagnostics, syntaxCheckDelimiters, lineNrChange);
+            DelayInvalidateOrUpdate(this.HeaderDiagnostics, this.UpdatedHeaderDiagnostics, syntaxCheckDelimiters, lineNrChange);
 
 
         /// <summary>
-        /// Replaces the current semantic diagnostics with the given sequence.
+        /// Replaces the current semantic diagnostics with the given sequence and pushes the updated semantic diagnostics.
```

Comment:
```suggestion
        /// Replaces both the current and updated semantic diagnostics with the given sequence.
```

It looks like this method is replacing both lists (unlike `AddAndFinalizeSemanticDiagnostics` which appends to the updated diagnostics).

Reply:
reworded